src/App.tsx

```diff
@@ -1,14 +1,70 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [theValue, setTheValue] = useState('all');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
+  }, []);
+
+  const visibleTodos = todos.filter(todo => {
+    const isQueryInclude = todo.title.toLowerCase().includes(searchQuery.toLowerCase().trim());
+    let isValue = true;
+
+    switch (theValue) {
+      case 'active':
+        isValue = !todo.completed;
+        break;
+
+      case 'completed':
+        isValue = todo.completed;
+        break;
+
+      default:
+        isValue = true;
+    }
+
+    return isQueryInclude && isValue;
+  });
```

Comment:
Let's move filter function to some helpers in separate file

Reply:
✔️